### PR TITLE
Resources, Service Accounts, labels and App Version

### LINKED ISSUE #2: [Bug]: Track App Version, not `latest`
### Contact Details

justinseiser@gmail.com

### What happened?

You almost certainly do not want to track the image `latest` since that will eventually cause issues when a node already has that image in place and then you guy's upgrade a release.

You would want to track:l https://github.com/memgraph/helm-charts/blob/main/charts/memgraph/Chart.yaml#L6
instead of `latest` https://github.com/memgraph/helm-charts/blob/main/charts/memgraph/values.yaml#L1

### Chart type

Standalone

### Chart version

0.1.0

### Environment

Local Machine

### Relevant log output

_No response_

### LINKED ISSUE #3: [Bug]: Label Usage
### Contact Details

justinseiser@gmail.com

### What happened?

1. Should use your templated labels

labels, and selectorLabels are templated:

https://github.com/memgraph/helm-charts/blob/main/charts/memgraph/templates/_helpers.tpl#L33

Yet they are not used: https://github.com/memgraph/helm-charts/blob/main/charts/memgraph/templates/statefulset.yaml#L12

2. Should allow setting custom labels

It would also be nice, to be able to add custom labels.

### Chart type

Standalone

### Chart version

0.1.0

### Environment

Local Machine

### Relevant log output

_No response_

### LINKED ISSUE #4: [Bug]: Service Account
### Contact Details

justinseiser@gmail.com

### What happened?

Should be able to define and configure a custom ServiceAccount.

Templating exists: https://github.com/memgraph/helm-charts/blob/main/charts/memgraph/templates/_helpers.tpl#L53

Just need to add the serviceAccount, and then add it to the statefulset

### Chart type

Standalone

### Chart version

0.1.0

### Environment

Local Machine

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jseiser**
fix(resources): Allow Setting Pod resources
fix(service_account): Allow Creating Custom Service Account
fix(labels): Use Label templates
fix(app_version): Use and default to app version instead of latest

Closes: #6 
Closes #4 
Closes #3 
Closes #2

### CODE REVIEW

**Comment by jseiser** [comment]
```
❯ helm template charts/memgraph | kubeconform --summary 
Summary: 6 resources found parsing stdin - Valid: 6, Invalid: 0, Errors: 0, Skipped: 0
```

File: charts/memgraph/values.yaml [hunk 1]
```diff
@@ -1,4 +1,8 @@
-image: memgraph/memgraph:latest
+image:
+  repository: memgraph/memgraph
+  # Overrides the image tag whose default is v{{ .Chart.AppVersion }}
+  tag: ""
```
Comment (alex-linx) [inline charts/memgraph/values.yaml:4]: I think the default should match the appVersion, so 2.10.0 in this case

**Comment by alex-linx** [comment]
@jseiser Awesome work, thanks!
Just a small thing in #6 I also asked to have the option for annotations on the statefulset / service.
If you could add it that would be amazing, if not I'll just open another issue/PR

File: charts/memgraph/values.yaml (hunk 1 above)
Comment (jseiser) [reply charts/memgraph/values.yaml:4]: That is what is already taking place

```
      containers:
        - name: memgraph
          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
```

We default to the app version if they do not explicitly set a image tag.

**Comment by jseiser** [comment]
> @jseiser Awesome work, thanks! Just a small thing in #6 I also asked to have the option for annotations on the statefulset / service. If you could add it that would be amazing, if not I'll just open another issue/PR

```
podAnnotations
ServiceAccount.annotations
service.annotations
```

File: charts/memgraph/templates/statefulset.yaml [hunk 2]
```diff
@@ -4,53 +4,60 @@ kind: StatefulSet
 metadata:
   name: {{ include "memgraph.fullname" . }}
   labels:
-    app.kubernetes.io/name: memgraph
-    app.kubernetes.io/managed-by: Helm 
+    {{- include "memgraph.labels" . | nindent 4 }}
+  {{- with .Values.podAnnotations }}
```
Comment (alex-linx) [inline charts/memgraph/templates/statefulset.yaml:8]: This should me moved under spec.template.metadata.annotations,
and then have a separate statefulSet.annotations values

**Comment by jseiser** [comment]
Sorry, moving too fast. Ill get this fixed in the morning

**Comment by jseiser** [comment]
@antejavor 

Anything else you need me to do on this one?  I know we are sitting in different timezones.

**Comment by antejavor** [comment]
@jseiser No at the moment, the changes look nice and are highly appreciated. Test charts failed for some reason, will look into it. We will probably merge this next week, and make a new release.

**Review by antejavor — APPROVED** [review]
This looks nice 🙌. Thanks for the contribution 💪, the CI fails for some reason, but I think it is due to an HTTP request. 
I will look into this Github action and add a pre-commit hook in the next PR. Thanks for opening the issue: https://github.com/memgraph/helm-charts/issues/8

Not sure if you are on our discord: https://discord.gg/memgraph, if you are would love to have a private DM about your needs for Kubernetes + Memgraph. 

Just ping me there.

**Pull request merged**